library/src/main/kotlin/com/vinted/actioncable/client/kotlin/Subscriptions.kt

```diff
@@ -21,8 +21,15 @@ class Subscriptions constructor(private val consumer: Consumer) {
      * @param channel Channel to connect
      * @return Subscription instance
      */
-    fun create(channel: Channel): Subscription = Subscription(consumer, channel).also {
-        subscriptions.add(it)
+    fun create(channel: Channel): Subscription {
+        require(channel.identifier !in subscriptions.map { it.identifier }) { "Such subscription already exists" }
```

Comment:
Typically libraries are intended to be thread-safe. Your ir not. Two threads could create a subscription for the same channel.